Simplified/Accounts/NYPLAgeCheck/NYPLAgeCheck.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+
+protocol NYPLAgeCheckCompletionDelegate: class {
+  func ageCheckCompleted(_ birthYear: Int)
+  func ageCheckFailed()
+}
+
+@objc protocol NYPLAgeCheckVerification {
+  func verifyCurrentAccountAgeRequirement(_ completion: ((Bool) -> ())?) -> Void
+}
+
+@objcMembers final class NYPLAgeCheck : NSObject, NYPLAgeCheckCompletionDelegate, NYPLAgeCheckVerification {
+  // Static methods and vars
+  static let sharedInstance = NYPLAgeCheck()
+
+  @objc class func shared() -> NYPLAgeCheck
+  {
+    return NYPLAgeCheck.sharedInstance
+  }
+  
+
+  // Members
+  let serialQueue = DispatchQueue(label: "\(Bundle.main.bundleIdentifier!).ageCheck")
+  var handlerList = [((Bool) -> ())]()
+  var isPresenting = false
+
+  func verifyCurrentAccountAgeRequirement(_ completion: ((Bool) -> ())?) -> Void
+  {
+    serialQueue.async {
+      let userAccount = NYPLUserAccount.sharedAccount()
+      guard let accountDetails = AccountsManager.shared.currentAccount?.details else {
+        completion?(false)
+        return
+      }
+      
+      if userAccount.needsAuth == true || accountDetails.userAboveAgeLimit {
+        completion?(true)
+        return
+      }
+      
+      if !accountDetails.userAboveAgeLimit && NYPLSettings.shared.userPresentedAgeCheck {
```

Comment:
Added a protocol named `NYPLAgeCheckChoiceStorage `.

I make it a parameter in the initializer and store it as a property in `NYPLAgeCheck` since we will be updating the `userPresentedAgeCheck ` property when AgeCheck completes/fails.